frontend/src/shared/util/hooks.ts

```diff
@@ -0,0 +1,36 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { useCallback, useEffect, useState } from 'react';
+
+export function useBackgroundRefresh (action: () => void, deps: readonly unknown[] = [], condition = true): boolean {
+    const [isBackgroundRefreshing, setIsBackgroundRefreshing] = useState(false);
+    const callbackAction = useCallback(action, [action, ...deps]);
+    
+    useEffect(() => {
+        const timerId = setInterval(() => {
+            if (condition) {
+                setIsBackgroundRefreshing(true);
+                callbackAction();
+            }
+        }, 60000);
```

Comment:
Any possible reason we might want this configurable? I could see a fourth parameter that could control the interval duration.

Reply:
I personally don't see the need to make it configurable but maybe @KyleLilly has more context on what would be more beneficial for the customer. We could potentially make this configurable in constants.ts or similar config file.

Reply:
It was just an idea because this is a generic hook I could  see it being used other places that might want something different. But if you don't feel there is a need we can let it stand as-is. It would be super easy in the future to change it if is needed.

Reply:
Yeah - this should really be configurable. Something that should probably be tied in with the in-app configuration work being done @douglas1850 .